### ISSUE TITLE
Message sent to me was not received

### ISSUE THREAD

**Issue by abranches** [emesene/emesene#482]
I noticed that sometimes I don't receive the messages sent to me by my contacts.
I was talking to a buddy and saw received messages logs in the debug window, that I didn't received in the conversation window.

Here's the debug log:

```
(17:59:34) gui.base.ConversationManager:on_contact_left: conversation is None
(17:59:34) gui.base.ConversationManager:on_contact_left: conversation is None
(17:59:36) papyon.transport:<<< RNG 1181572109 256.256.256.256:1863 CKI 1157253.8338110 myfriendemailaddress@hotmail.com André U messenger.msn.com 1
(17:59:36) papyon.protocol.switchboard:New switchboard session 1181572109
(17:59:36) papyon.transport:<-> Connecting to 256.256.256.256:1863
(17:59:36) papyon.transport:>>> ANS 0 myownemaileaddress@hotmail.com;{d12688d7-c094-414e-80d1-45fbe0894942} 1157253.8338110 1181572109
(17:59:36) papyon.transport:<<< IRO 0 1 2 myfriendemailaddress@hotmail.com;{f5a1b6ba-4d86-40aa-9b5b-4f83c6351f6c} André 2754609424:3624129584
(17:59:36) papyon.transport:<<< IRO 0 2 2 myfriendemailaddress@hotmail.com André 2754609424:3624129584
(17:59:36) papylib.Events:Contact's client capabilities changed: <papyon.Contact id='0be884bb-9f47-4380-8248-15f0176bc1a8' network='1' account='myfriendemailaddress@hotmail.com' memberships='FORWARD | ALLOW | REVERSE'>
(17:59:36) papyon.transport:<<< ANS 0 OK
(17:59:36) papyon.protocol.switchboard_manager:Handler <SwitchboardConversation participants="myfriendemailaddress@hotmail.com"> attached to switchboard 1181572109
(17:59:36) papyon.transport:<<< JOI myownemaileaddress@hotmail.com º%20Pedro 2754609424:2550273072
(17:59:37) papyon.transport:<<< MSG myfriendemailaddress@hotmail.com André 179
    X-MMS-IM-Format: 'FN=Sans; EF=; CO=000000; CS=1; PF=00'\r\n
    Content-Type: 'text/plain; charset=utf-8'\r\n
    MIME-Version: '1.0'\r\n
    \r\n
    bem, acho que brinco mais amanh\xc3\xa3
(17:59:37) e3.common.Signal:Signal handler (_on_message._on_message) error: 'NoneType' object has no attribute 'message_waiting'
(17:59:37) e3.common.Signal:Signal handler (_on_message._on_message) error: 1304268660.949675
```

The message that appears in the debug didn't show up in the conversation.
It happened with one more message right next to this one and after that every message was received normally.

I am using the current development version and my contact was using emesene 2.11.4 .

**Comment by c10ud** [comment]
interesting, seems like the conversation is dead.... mmm

2011/5/1 abranches
reply@reply.github.com:

**Comment by abranches** [comment]
actually it isn't the first time that I noticed this happening. but I thought it was a problem with a more buggy version of emesene(the one in ubuntu repos). this was the first time I could "catch" the problem.

**Comment by c10ud** [comment]
this happens on latest git, right? did you close window or whatever?
seems like something in the code thinks the conversation is still
alive while it's not

2011/5/1 abranches
reply@reply.github.com:

**Comment by abranches** [comment]
yes, latest git. I didn't do anything really :p. it simply happened while the conversation was going.

**Comment by Sbte** [comment]
It would be helpful if you had a way to reproduce it.

**Comment by abranches** [comment]
hmm... so strange! I went to see the conversation history now and that message was there.. I'm not getting nuts :p . it should be something in the way the conversation window receives the message from papyon (really don't know how it works..)

**Comment by c10ud** [comment]
well, we just tell to the conversation with a given id to get the
message..in your debug trace the id is dead, but if it's dead it
should have created a new one, that's the strange thing

2011/5/1 abranches
reply@reply.github.com:

**Comment by deep34** [comment]
hi all, 
yesterday i updated ubuntu to 11.04, I use now emesene 2.
I have the same problem as abranches, i don't receive some messages sent to me.
I will post later the logs if i succeed to reproduce the bug.

Good job guys !

**Comment by deep34** [comment]
Here's the debug log:

(17:56:29) papyon.transport:>>> MSG 5 N 212

```
X-MMS-IM-Format: 'FN=Sans; EF=; CO=000000; CS=1; PF=00'\r\n
Content-Type: 'text/plain; charset=utf-8'\r\n
MIME-Version: '1.0'\r\n
\r\n
my message 
```

(17:56:29) papyon.transport:<<< MSG myadress@msn.com myadress@msn.com 212

```
X-MMS-IM-Format: 'FN=Sans; EF=; CO=000000; CS=1; PF=00'\r\n
Content-Type: 'text/plain; charset=utf-8'\r\n
MIME-Version: '1.0'\r\n
\r\n
my message
```

(17:56:53) papyon.protocol.switchboard:Switchboard timed out. Going to leave it.
(17:56:53) papyon.protocol.switchboard:Leaving switchboard 1825797142.50178248.150188212
(17:56:53) papyon.transport:>>> OUT 6
(17:56:53) papyon.protocol.switchboard:Disconnected (1825797142.50178248.150188212)
(17:56:53) papyon.protocol.switchboard_manager:Closing switchboard handler <SwitchboardConversation participants="friend@hotmail.com">
(17:56:53) papyon.transport:<<< BYE myadress@msn.com;{7240d69a-81d0-4d79-9150-97d5dde9511e}
(17:57:06) papyon.transport:<<< MSG friend@hotmail.com Hatim 123

```
TypingUser: 'friend@hotmail.com'\r\n
Content-Type: 'text/x-msmsgscontrol'\r\n
MIME-Version: '1.0'\r\n
\r\n
\r\n
```

(17:57:11) papyon.transport:<<< MSG friend@hotmail.com Hatim 123

```
TypingUser: 'friend@hotmail.com'\r\n
Content-Type: 'text/x-msmsgscontrol'\r\n
MIME-Version: '1.0'\r\n
\r\n
\r\n
```

(17:57:13) papyon.transport:<<< MSG friend@hotmail.com Hatim 174

```
X-MMS-IM-Format: 'FN=Segoe%20UI; EF=; CO=0; CS=0; PF=22'\r\n
Content-Type: 'text/plain; charset=UTF-8'\r\n
MIME-Version: '1.0'\r\n
\r\n
message not received 1
```

(17:57:21) papyon.transport:<<< MSG friend@hotmail.com Hatim 123

```
TypingUser: 'friend@hotmail.com'\r\n
Content-Type: 'text/x-msmsgscontrol'\r\n
MIME-Version: '1.0'\r\n
\r\n
\r\n
```

(17:57:23) papyon.transport:<<< MSG friend@hotmail.com Hatim 154

```
X-MMS-IM-Format: 'FN=Segoe%20UI; EF=; CO=0; CS=0; PF=22'\r\n
Content-Type: 'text/plain; charset=UTF-8'\r\n
MIME-Version: '1.0'\r\n
\r\n
message not received 2
```

(17:58:01) papyon.transport:<<< UBX 1:friend@hotmail.com 568

```
[payload]
```

(17:58:01) papyon.profile:Set contact friend@hotmail.com flags to 1
(17:58:01) papyon.transport:<<< UBX 1:friend@hotmail.com 568

```
[payload]
```

(17:58:01) papyon.profile:Set contact friend@hotmail.com flags to 1
(17:58:06) papyon.transport:<<< UBX 1:friend@hotmail.com 577

```
[payload]
```

(17:58:06) papyon.profile:Set contact friend@hotmail.com flags to 1
(17:58:06) papyon.transport:<<< UBX 1:friend@hotmail.com 577

```
[payload]
```

(17:58:06) papyon.profile:Set contact friend@hotmail.com flags to 1
(17:58:23) papyon.protocol.switchboard:Switchboard timed out. Going to leave it.
(17:58:23) papyon.protocol.switchboard:Leaving switchboard 1825797142
(17:58:23) papyon.transport:>>> OUT 1
(17:58:23) papyon.protocol.switchboard:Disconnected (1825797142)
(17:58:23) papyon.protocol.switchboard_manager:Closing switchboard handler <SwitchboardConversation participants="friend@hotmail.com">
(17:59:19) papyon.transport:<<< NLN IDL 1:anotherContact@msn.com Dam 3055849776:2609258384 %3cmsnobj%20Creator%3d%22anotherContact%40msn.com%22%20Type%3d%223%22%20SHA1D%3d%22y7V2SirqrDBNP%2fo1nE6LmNmTL4Y%3d%22%20Size%3d%2225966%22%20Location%3d%220%22%20Friendly%3d%22MgAwADEAMQAtADAANAAtADAAOQAgADEANwAtADEAMAAtADUAOQAuADMANAA4AAAA%22%2f%3e
(17:59:19) papylib.Events:Contact's client capabilities changed: <papyon.Contact id='22c99a92-f169-4f18-92b5-d64f89b2af43' network='1' account='anotherContact@msn.com' memberships='FORWARD | ALLOW | REVERSE'>
(17:59:20) papyon.transport:<<< NLN IDL 1:anotherContact@msn.com Dam 3055849776:2609258384 %3cmsnobj%20Creator%3d%22anotherContact%40msn.com%22%20Type%3d%223%22%20SHA1D%3d%22y7V2SirqrDBNP%2fo1nE6LmNmTL4Y%3d%22%20Size%3d%2225966%22%20Location%3d%220%22%20Friendly%3d%22MgAwADEAMQAtADAANAAtADAAOQAgADEANwAtADEAMAAtADUAOQAuADMANAA4AAAA%22%2f%3e
(17:59:20) papylib.Events:Contact's client capabilities changed: <papyon.Contact id='22c99a92-f169-4f18-92b5-d64f89b2af43' network='1' account='anotherContact@msn.com' memberships='FORWARD | ALLOW | REVERSE'>

**Comment by c10ud** [comment]
interesting trace.did you have more than one conversation opened?

2011/5/3 deep34
reply@reply.github.com:

**Comment by deep34** [comment]
yes 3

**Comment by c10ud** [comment]
@abranches if you were trying to fix this issue, please be aware this mightn't be a bug in emesene but in python-papyon instead. seems like a closed switchboard isn't so closed after all -- while emesene thinks it's closed (as it should be though).

http://lists.freedesktop.org/archives/papyon/2011-May/000030.html

**Comment by abranches** [comment]
ok c10ud. i'll keep following the discussion at the papyon mailling list :)

**Comment by arielj** [comment]
Is this finally a papyon bug? can we close this?

**Comment by MasterJ0k3R** [comment]
I get this problem when I have two sessions open. In particular,  when I have a WLM (from hotmail webpage) and an emesene session at the same time. At the begining, both sessions work fine but after a few minutes only emesene starts to miss all the messages from the other person.

**Comment by hit** [comment]
I also got some debug:

DEBUG:papyon.transport:>>> MSG 7 N 189
    X-MMS-IM-Format: 'FN=Verdana; EF=B; CO=000000; CS=1; PF=00'\r\n
    Content-Type: 'text/plain; charset=utf-8'\r\n
    MIME-Version: '1.0'\r\n
    \r\n
    [SOME TEXT HERE]
DEBUG:papyon.transport:<<< MSG rene@[MAIL] R 189
    X-MMS-IM-Format: 'FN=Verdana; EF=B; CO=000000; CS=1; PF=00'\r\n
    Content-Type: 'text/plain; charset=utf-8'\r\n
    MIME-Version: '1.0'\r\n
    \r\n
    [SOME TEXT HERE TOO, TWICE, but only twice in log not in conversation]
INFO:papyon.protocol.switchboard:Switchboard timed out. Going to leave it.
INFO:papyon.protocol.switchboard:Leaving switchboard 231084747.20734115.6244220
DEBUG:papyon.transport:>>> OUT 8
DEBUG:papyon.transport:<<< BYE rene@[MAIL];{b6f49b6f-cda5-4695-93ee-73be6e3d10ed}
INFO:papyon.protocol.switchboard:Disconnected (231084747.20734115.6244220)
INFO:papyon.protocol.switchboard_manager:Closing switchboard handler <SwitchboardConversation participants="geitlin@[MAIL]">
DEBUG:papyon.transport:<<< MSG geitlin@[MAIL] geitlin 114
    TypingUser: 'geitlin@[MAIL]'\r\n
    Content-Type: 'text/x-msmsgscontrol'\r\n
    MIME-Version: '1.0'\r\n
    \r\n
    \r\n
DEBUG:papyon.transport:<<< MSG geitlin@[MAIL] geitlin 148
    X-MMS-IM-Format: 'FN=Trebuchet%20MS; EF=; CO=0; CS=ba; PF=22'\r\n
    Content-Type: 'text/plain; charset=UTF-8'\r\n
    MIME-Version: '1.0'\r\n
    \r\n
    [SOME ANSWER HERE, which did not show up in conversation]

I told something and contact answered almost instantly, so no timeout should happen, but still I couldn't see her message in conversation. If it's only papyon-related, I'll file a bug to bugzilla.

**Comment by c10ud** [comment]
yes, it's papyon related, and it's related to the patch that makes the switchboard close after 60 secs of inactivity..seems like something went wrong with the timeout because 60secs didn't pass from what you're saying.. also, papyon didn't reopen a new switchboard when your friend replied! double bug (!)

**Comment by c10ud** [comment]
this should have been fixed in current papyon-git-master. expect this issue to be fixed in papyon-0.5.6 (still unreleased)

probably we will make 2.11.8 dependent on that version
